elftools/dwarf/compileunit.py

```diff
@@ -55,6 +56,8 @@ def __init__(self, header, dwarfinfo, structs, cu_offset, cu_die_offset):
 
         # A list of DIEs belonging to this CU. Lazily parsed.
         self._dielist = []
+        # A list of corresponding DIE offsets.
+        self._diemap = []
```

Comment:
Can you please add a substantial comment here explaining how the laziness works. What are the invariants for the relevant method calls, and how _dielist and _diemap interplay?